consensus/istanbul/backend/backend.go

```diff
@@ -251,6 +255,27 @@ func (sb *Backend) Commit(proposal istanbul.Proposal, bitmap *big.Int, seals []b
 	if err != nil {
 		return err
 	}
+
+	parent := sb.chain.GetHeader(h.ParentHash, h.Number.Uint64()-1)
+	// If the parent was not the genesis block then try to use the parent's
+	// extras as the seal for this block
+	if parent != nil {
+		parentExtras, err := types.ExtractIstanbulExtra(parent)
+		if err != nil {
+			return err
+		}
+
+		block = block.WithParentSeal(
+			&types.BlockSeal{
+				Bitmap: parentExtras.Bitmap,
+				Seal:   parentExtras.CommittedSeal,
+			},
+		)
+	} else {
+		// otherwise initialize with default parent seal
+		block = block.WithParentSeal(&types.EmptyBlockSeal)
```

Comment:
I'm assuming this is illegal for all non-genesis blocks?

Reply:
Correct